tutorials/holoscan_debugging/README.md

```diff
@@ -0,0 +1,302 @@
+As of v2.3 we do not release Holoscan SDK builds of this type, but we do provide a `run` script to help
+developers generate their own `debug` builds on demand.
+
+## What are some common tools I can use for debugging my application?
+
+There are a wide variety of free and/or open source software tools available for general C++ and Python debugging, including:
+- [NVIDIA's debugging solutions](https://developer.nvidia.com/debugging-solutions), including NVIDIA NSight and CUDA-GDB;
+- The [GNU project DeBugger (GDB)]([GDB](https://sourceware.org/gdb/));
+- The built-in [Python Debugger (pdb)](https://docs.python.org/3/library/pdb.html) module;
+- [Microsoft Visual Studio Code](https://code.visualstudio.com/), with a wide variety of community extensions.
+
+In this tutorial we will focus on the "headless" `GDB` and `pdb` tools. These require minimal setup and can be run via a simple
+terminal without a dedicated display. For advanced development we recommend reviewing Visual Studio Code Development Containers
+and debug profiles as used in several HoloHub applications.
+
+### References
+
+To get started with debugging your Holoscan SDK application, visit the [Debugging](https://docs.nvidia.com/holoscan/sdk-user-guide/holoscan_debugging.html) user guide section for common topics including:
+- Generating debug symbols with VSCode
+- Live debugging for C++ and Python applications
+- Inspecting application crashes
+- Profiling and code coverage
+
+Visit the [Logging](https://docs.nvidia.com/holoscan/sdk-user-guide/holoscan_logging.html) user guide
+section for a thorough overview on how to set up Holoscan SDK logging in your C++ or Python application.
+
+## Debugging a C++ Application with `gdb`
+
+### Background
+
+[GDB](https://sourceware.org/gdb/) (the GNU project DeBugger) is a widely used tool for headless just-in-time debugging of C++ applications. GDB distributions are available for most Holoscan SDK supported platforms and do not require a display to set up. Refer to the [GDB User Manual](https://sourceware.org/gdb/current/onlinedocs/gdb.html/)
+to get started.
+
+### Prerequisites
+
+The steps for getting started with `gdb` depend on how you are consuming Holoscan SDK.
+- We encourage using Holoscan SDK containers from NGC for development and we take this approach for most of the tutorial. If you are using an NGC container for __Holoscan SDK v2.3.0 or later__, you already have access
+  to debug symbols and can get started right away.
+- If you are using an older Holoscan SDK container from NGC, or if you are consuming Holoscan SDK through another means such as Debian packages, Python wheels, or custom installation, you will need to build
+Holoscan SDK with debugging symbols in order to step through Holoscan SDK code during debugging. Jump to the [Legacy Holoscan SDK](#debugging-symbols-for-legacy-holoscan-sdk-versions) section to get started.
+
```

Comment:
Aren't these prerequisites true independent of which tool to use (gdb, pdb, legacy)?
I'd suggest moving these to end of section "What are debugging symbols?", as general per-requisites.

Reply:
Agreed, I've now moved these earlier in the tutorial as a general "Prerequisites" section